### PR TITLE
fix: updated setup.ts script to configure and verify SMTP credentials in env file during setup

### LINKED ISSUE #1606: Setup script must prompt for SMTP parameters
**Is your feature request related to a problem? Please describe.**

The installation script should have greater functionality when configuring SMTP Email parameters

**Describe the solution you'd like**

1. The installation script must prompt for each of these parameters
    ```
    IS_SMTP=
    SMTP_HOST=
    SMTP_PASSWORD=
    SMTP_USERNAME=
    SMTP_PORT=
    SMTP_SSL_TLS=
    ```
1. If the installer does not want to configure SMTP, then the rest of the SMTP parameters must be skipped.
1. The installer may not know what SMTP is, therefore the prompt should let them know that SMTP configuration is necessary for Talawa to send Email
1.  The script should verify that SMTP connectivity is present and that authentication works given the parameters provided

**Describe alternatives you've considered**
- N/A

**Approach to be followed (optional)**
- N/A

**Additional context**
- N/A

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship:
- https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by MahendraDani**
**What kind of change does this PR introduce?**
Feature

**Issue Number:**

Fixes #1606 

**Did you add tests for your changes?**
No

**Snapshots/Videos:**

https://github.com/PalisadoesFoundation/talawa-api/assets/116940083/37fb6a7d-c475-4006-bcc0-f8690696158e

- Case 1: User configures SMTP with correct credentials

https://github.com/PalisadoesFoundation/talawa-api/assets/116940083/d031ff82-d873-4073-8a35-22bdaa2b5eb3

- Case 2: Users skips to configure SMTP

**If relevant, did you update the documentation?**

**Summary**
Added script to configure SMTP environment variables during setup

**Does this PR introduce a breaking change?**
No

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

**Comment by MahendraDani** [comment]
@palisadoes If there are any changes please let me know

**Comment by MahendraDani** [comment]
https://github.com/PalisadoesFoundation/talawa-api/assets/116940083/dd218782-3494-4fb0-8705-d0dcce468a11

- Case 3: When users configures SMTP with incorrect credentials

File: setup.ts [hunk 1]
```diff
@@ -404,6 +405,107 @@ async function importData(): Promise<void> {
...
+  ]);
+
+  const isValidSmtpConfig =
+    smtpConfig.SMTP_HOST &&
+    smtpConfig.SMTP_PORT &&
+    smtpConfig.SMTP_USERNAME &&
+    smtpConfig.SMTP_PASSWORD;
+
+  if (!isValidSmtpConfig) {
+    console.error(
```
Comment (sumitra19jha) [inline setup.ts:484]: There can be multiple reason for failuer. Pleas add error in the log and also make sure not to log password if exists in error

**Review by sumitra19jha — APPROVED** [review]

File: setup.ts (hunk 1 above)
Comment (MahendraDani) [reply setup.ts:484]: Sure

Comment (MahendraDani) [reply setup.ts:489]: @sumitra19jha 
Added error message in the log

https://github.com/PalisadoesFoundation/talawa-api/assets/116940083/c774b6ad-9721-42ca-bcad-561c45e731e4

**Comment by palisadoes** [comment]
Please fix the failing tests

**Comment by MahendraDani** [comment]
> Please fix the failing tests

@palisadoes I am not able to understand the cause of the failing tests

**Comment by MahendraDani** [comment]
> > Please fix the failing tests
> 
> @palisadoes I am not able to understand the cause of the failing tests
Because my last commit had no failing tests, then I merged the `develop` branch with this branch and then the linting, formating test is failing

**Comment by palisadoes** [comment]
Please read the error message in the failing test. It is explained there

**Comment by palisadoes** [comment]
@NamitBhutani 
The linting is failing for this PR and probably all new PR updates. Please rectify quickly or we will need to revert your PR:

- https://github.com/PalisadoesFoundation/talawa-api/pull/1635

Context

1. https://github.com/PalisadoesFoundation/talawa-api/pull/1647
2. https://github.com/PalisadoesFoundation/talawa-api/actions/runs/7442918033/job/20247042509?pr=1647

**Comment by MahendraDani** [comment]
> @NamitBhutani The linting is failing for this PR and probably all new PR updates. Please rectify quickly or we will need to revert your PR:
> 
> * [Lint only updated files with stricter linting #1635](https://github.com/PalisadoesFoundation/talawa-api/pull/1635)
> 
> Context
> 
> 1. [fix: updated setup.ts script to configure and verify SMTP credentials in env file during setup #1647](https://github.com/PalisadoesFoundation/talawa-api/pull/1647)
> 2. https://github.com/PalisadoesFoundation/talawa-api/actions/runs/7442918033/job/20247042509?pr=1647

@palisadoes Should I wait until the above is resolved?

**Comment by palisadoes** [comment]
It looks that way.

**Comment by palisadoes** [comment]
I'm going to revert the other PR. It's not working as expected

**Comment by MahendraDani** [comment]
> I'm going to revert the other PR. It's not working as expected

Ok sir.
I hope my PR is clean and does not cause the error

**Comment by palisadoes** [comment]
@MahendraDani

Please merge with the latest upstream. And update this PR with a commit. It should be fixed now

**Comment by MahendraDani** [comment]
> @MahendraDani
> 
> Please merge with the latest upstream. And update this PR with a commit. It should be fixed now

Yes sir

**Comment by MahendraDani** [comment]
@palisadoes Merged with latest upstream and the previously failing test is now passing

**Comment by MahendraDani** [comment]
@sumitra19jha Would you please review the PR I have made the changes suggested by you.

**Comment by sumitra19jha** [comment]
In error.response is it possible the password can be inputed, if yes, make sure you hide it. Just check whatever password is entered is replace by some string

**Comment by MahendraDani** [comment]
> In error.response is it possible the password can be present, if yes, make sure you hide it. Just check whatever password is entered is replace by some string if present in error response

@sumitra19jha I have checked that. The password will never be shown in error response

**Comment by MahendraDani** [comment]
https://github.com/PalisadoesFoundation/talawa-api/assets/116940083/255782b8-4d17-483a-9061-65f741f252af

@sumitra19jha You can observe in above recording that the error is caused due to incorrect password and the password is not shown in error message

**Comment by sumitra19jha** [comment]
@palisadoes good to merge

**Pull request merged**